apps/etterlatte-behandling/src/test/kotlin/behandling/klage/KlageRoutesIntegrationTest.kt

```diff
@@ -0,0 +1,159 @@
+                addAuthToken(fagsystemTokenEY)
+                contentType(ContentType.Application.Json)
+                setBody(EgenAnsattSkjermet(fnr = fnr, inntruffet = Tidspunkt.now(), skjermet = true))
+            }
+
+            // henter med saksbehandler som mangler tilgang
+            val responseNotFound = client.get("/api/klage/${klage.id}") {
+                addAuthToken(tokenSaksbehandler)
+            }
+            assertEquals(HttpStatusCode.NotFound, responseNotFound.status)
+        }
+    }
+
+    @Test
+    fun `opprettelse av klage går bra og henting gjør tilgangskontroll når saken får adressebeskyttelse`() {
+        withTestApplication { client ->
+            val fnr = Folkeregisteridentifikator.of("08071272487").value
+            val sak: Sak = client.post("/personer/saker/${SakType.BARNEPENSJON}") {
+                addAuthToken(tokenSaksbehandler)
+                contentType(ContentType.Application.Json)
+                setBody(FoedselsnummerDTO(fnr))
+            }.apply {
+                assertEquals(HttpStatusCode.OK, status)
+            }.body()
+            val klage: Klage = client.post("/api/klage/opprett/${sak.id}") {
+                addAuthToken(tokenSaksbehandler)
+            }.body()
+            val response = client.get("/api/klage/${klage.id}") {
+                addAuthToken(tokenSaksbehandler)
+            }
+            assertEquals(HttpStatusCode.OK, response.status)
+            val hentetKlage = response.body<Klage>()
+            assertEquals(klage, hentetKlage)
+
+            // Setter strengt fortrolig på saken klagen hører til
+            client.post("/grunnlagsendringshendelse/adressebeskyttelse") {
+                addAuthToken(fagsystemTokenEY)
+                contentType(ContentType.Application.Json)
+                setBody(
+                    Adressebeskyttelse(
+                        hendelseId = "1",
+                        endringstype = Endringstype.OPPRETTET,
+                        fnr = fnr,
+                        adressebeskyttelseGradering = AdressebeskyttelseGradering.STRENGT_FORTROLIG
+                    )
+                )
+            }
+
+            val responseVanligSaksbehandler = client.get("/api/klage/${klage.id}") {
+                addAuthToken(tokenSaksbehandler)
+            }
+            assertEquals(HttpStatusCode.NotFound, responseVanligSaksbehandler.status)
+            val responseSaksbehandlerMedTilgang = client.get("/api/klage/${klage.id}") {
+                addAuthToken(tokenSaksbehandlerMedStrengtFortrolig)
+            }
+            assertEquals(HttpStatusCode.OK, responseSaksbehandlerMedTilgang.status)
+            assertEquals(
+                klage.copy(sak = klage.sak.copy(enhet = Enheter.STRENGT_FORTROLIG.enhetNr)),
+                responseSaksbehandlerMedTilgang.body<Klage>()
+            )
+        }
+    }
+
+    private fun withTestApplication(block: suspend (client: HttpClient) -> Unit) {
```

Comment:
Lukter noe common greier her

Reply:
det er ikke skrevet noe common for det, og tar ikke det nå heller :D